docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,287 @@
+    // Therefore, costs above 30 require the use of bookshelves or other enchantment boosting blocks to be obtainable in survival.
+
+    "min_cost": {
+        "base": 2,
+        "per_level_above_first": 8
+    },
+    // Specifies the minimum cost of this enchantment; see above.
+
+    "anvil_cost": 2,
+    // The cost that this enchantment adds to repairing an item in an anvil in levels. The cost is multiplied by enchantment level.
+    // If an item has a DataComponentTypes.STORED_ENCHANTMENTS component, the cost is halved. In vanilla, this only applies to enchanted books.
+    // Bounded by [1, inf).
+
+    "slots": [
+        "mainhand"
+    ],
+    // (Optional) A list of slot groups this enchantment provides effects in. 
+    // A slot group is defined as one of the possible values of the EquipmentSlotGroup enum.
+    // In vanilla, these are: `any`, `hand`, `mainhand`, `offhand`, `armor`, `feet`, `legs`, `chest`, `head`, and  `body`.
+
+    "effects": {
+    "examplemod:custom_effect": [
+      {
+        "effect": {
+          "type": "minecraft:add",
+          "value": {
+            "type": "minecraft:linear",
+            "base": 1,
+            "per_level_above_first": 1
+          }
+        }
+      }
+    ]
+  }
+  // The effects that this enchantment provides as a map of Enchantment Effect Components.
+}
+```
+
+## Enchantment Effect Components
+Enchantment Effect Components are specially-registered [Data Components] that determine how an enchantment functions. The type of the component defines its effect, while the data it contains is used to inform or modify that effect. For instance, the `minecraft:damage` component modifies the damage that a weapon deals by an amount determined by its data.
+
+Vanilla defines various [built-in enchantment effect components], which are used to implement all vanilla enchantments.
+
+### Custom Enchantment Effect Components
+Enchantment Effect Component Types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`, which takes a `DataComponentType<?>`. For example, one could register an Enchantment Effect Component that can store an `ExampleEffectData` object as follows:
+
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ExampleEffectData>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<ExampleEffectData>builder()
+            .persistent(ExampleEffectData.CODEC)
+            .build());
+```
+
+There are no inheritance requirements on the data held by an Enchantment Effect Component, but it may be helpful to refer to vanilla to ensure compatibility with the vanilla helper methods.
```

Comment:
Maybe show an example, even if there's no real requirements.

Reply:
Since I lead straight into a section where I'm discussing one of the most important of these interactions (the one with ConditionalEffect), and the others would require discussing classes that aren't mentioned outside the other entry, I think I might just cut this.

Reply:
Works for me too.